### PR TITLE
[HUSSEIN] 100% coverage on both function's tests

### PR DESCRIPTION
**Pull request by HusseinSarrar05**
# Overview
## My two functions are:
1. **upload** in **api/legacy.py**
2. **view_or_basicauth** in **userfeeds/auth.py**

## The tests for these functions, respectively, are in the following files:
1. **mygpo/api/tests.py**
2. **mygpo/userfeeds/tests.py**

# Results

Initially, both my functions had no tests written for them at all, and so both had 0% coverage from the original tests.
I have developed tests for these two functions and, through manual instrumentation, I am certain that I achieved 100% coverage for both.

The results of manual instrumentation are written to the **view_or_basicauth_coverage** and **legacy_coverage** **txt** files directly under the mother **mygpo** directory.

### CODE REVIEW

File: mygpo/api/legacy.py [hunk 1]
```diff
@@ -21,42 +21,97 @@
...
 
     i = Importer(opml)
 
-    podcast_urls = [p["url"] for p in i.items]
+    # podcast_urls = [p["url"] for p in i.items]
+
+    # Break down iteration
+    podcast_urls = []
+    for p in i.items:
+        # Branch ID: 4
```
Comment (spower77) [inline mygpo/api/legacy.py:66]: Missing the coverage[4] = true statement

File: mygpo/userfeeds/auth.py [hunk 2]
```diff
@@ -16,36 +18,74 @@ def view_or_basicauth(view, request, username, token_name, realm="", *args, **kw
...
+                    # Branch ID: 10
+                    branch_coverage[10] = True
                     return view(request, uname, *args, **kwargs)
+                else:
+                    # Branch ID: 11
+                    branch_coverage[11] = True
+            else:
+                # Branch ID: 12
+                branch_coverage[12] = True
+    else:
```
Comment (spower77) [inline mygpo/userfeeds/auth.py:88]: Not sure if this belongs to the first if statement or the second if statement. Missing a branch coverage for one of the two outer if statements

File: mygpo/web/forms.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 logger = logging.getLogger(__name__)
```
Comment (spower77) [inline mygpo/web/forms.py:11]: Flag

**Review by spower77 — CHANGES_REQUESTED** [review]
Please fix the two missing coverages

File: mygpo/api/legacy.py (hunk 1 above)
Comment (HusseinSarrar05) [reply mygpo/api/legacy.py:66]: Thats why this was never triggering, will fix

File: mygpo/userfeeds/auth.py (hunk 2 above)
Comment (HusseinSarrar05) [reply mygpo/userfeeds/auth.py:88]: You were right, I missed one of the else clauses

**Review by spower77 — APPROVED** [review]
Great. 100% coverage of selected functions achieved

**Pull request merged**